sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -647,15 +647,15 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * @since 2.0.0
    */
   override def refreshTable(tableName: String): Unit = {
-    val tableIdent = sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName)
-    val relation = sparkSession.table(tableIdent).queryExecution.analyzed
+    val relation = sparkSession.table(tableName).queryExecution.analyzed
 
     relation.refresh()
 
+    val ident = sparkSession.sessionState.sqlParser.parseMultipartIdentifier(tableName)
     // Temporary and global temporary views are not supposed to be put into the relation cache
     // since they are tracked separately.
-    if (!sessionCatalog.isTempView(tableIdent)) {
-      sessionCatalog.invalidateCachedTable(tableIdent)
+    if (!sessionCatalog.isTempView(ident)) {
+      sessionCatalog.invalidateCachedTable(makeTableIdentifier(ident))
```

Comment:
TableCatalog has its own `invalidateTable` so it seems we can not only do it with session catalog. An option may be: construct a `RefreshTable` and call `sessionState.executePlan(RefreshTable)`, so the analyzer can decide use which catalog, i.e. session catalog or other catalog.

Reply:
What is a different of `executePlan(RefreshTable).analyzed` and `executePlan(RefreshTable).rdd`? Which one triggers  to have `RefreshTableExec` created and executed?

Reply:
cc @cloud-fan what do you think of this idea?

Reply:
we should use `executePlan(RefreshTable).assertCommandExecuted`

Reply:
the idea LGTM, to fully support refresh table in any catalog.